base/error.jl

```diff
@@ -197,15 +197,16 @@ windowserror(p, code::UInt32=Libc.GetLastError(); extrainfo=nothing) = throw(Mai
 """
     @assert cond [text]
 
-Throw an [`AssertionError`](@ref) if `cond` is `false`. Preferred syntax for writing assertions.
+Throw an [`AssertionError`](@ref) if `cond` is `false`. Debugging tool for writing assertions.
 Message `text` is optionally displayed upon assertion failure.
-
+                                            
 !!! warning
     An assert might be disabled at various optimization levels.
     Assert should therefore only be used as a debugging tool
-    and not used for authentication verification (e.g., verifying passwords),
-    nor should side effects needed for the function to work correctly
-    be used inside of asserts.
+    and not used for authentication verification (e.g., verifying passwords).
+    The user must not rely on the return value of `@assert` for the correct behavior 
```

Comment:
The return value of `@assert true` is `nothing` (and should remain that way at all optimization levels)